File: leetcode/26.cpp
#include <vector>

using namespace std;

class Solution {
public:
  int removeDuplicates(vector<int> &nums) {
    if (nums.size() == 0) {
      return 0;
    }
    int len = 0;
    for (int num : nums) {
      if (num != nums[len]) {
        nums[++len] = num;
      }
    }
    return len + 1;
  }
};

File: leetcode/26.py
from typing import List


class Solution:

    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0
        cur = 0
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                cur += 1
                nums[cur] = nums[i]
        return cur + 1
